### ISSUE TITLE
[docs] `participantindex` default value not clear

### ISSUE THREAD

**Issue by goekce** [eclipse-cyclonedds/cyclonedds#1122]
I see two default values. Is this a bug or feature in the [`participantindex` docs](https://github.com/eclipse-cyclonedds/cyclonedds/blob/master/docs/manual/options.md#cycloneddsdomaindiscoveryparticipantindex)?:

https://github.com/eclipse-cyclonedds/cyclonedds/blob/0bf9552ab4569a813565ff03495bf4122a14814f/docs/manual/options.md?plain=1#L127-L129

**Comment by goekce** [comment]
> otherwise you won't even have communication within the machine.

Thanks for pointing that out @eboasson, this helped me significantly to fix the following problem in my ROS environment:

I have a group of computers that I put together in one Wireguard network. Wireguard can send the multicast messages only to one point-to-point link, so I had to switch to unicast networking. Then I configured CycloneDDS using the environment variable `CYCLONEDDS_URI` as follows:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<CycloneDDS xmlns="https://cdds.io/config" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://cdds.io/config https://raw.githubusercontent.com/eclipse-cyclonedds/cyclonedds/master/etc/cyclonedds.xsd">
    <Domain Id="any">
        <General>
            <NetworkInterfaceAddress>wg0</NetworkInterfaceAddress>
            <AllowMulticast>false</AllowMulticast>
        </General>
        <Discovery>
            <Peers>
                <Peer address='my_computer'></Peer>
                <Peer address='robot0'></Peer>
                <Peer address='robot1'></Peer>
            </Peers>
        </Discovery>
    </Domain>
</CycloneDDS>
```

Now the computers could see each others' topics. Unfortunately this led to another problem. Suddenly I got the following error messages when I launched Gazebo simulations. I'll paste them here as a potential reference for other TurtleBot3 users:

```
$ ros2 launch turtlebot3_gazebo turtlebot3_world.launch.py
...
[spawn_entity.py-4] [ERROR] [...] [spawn_entity]: Service %s/spawn_entity unavailable. Was Gazebo started with GazeboRosFactory?
[spawn_entity.py-4] [ERROR] [...] [spawn_entity]: Spawn service failed. Exiting.
[ERROR] [spawn_entity.py-4]: process has died [pid ..., exit code 1, cmd '.../lib/gazebo_ros/spawn_entity.py ...].
```

I could fix the problem by removing `<AllowMulticast>false</AllowMulticast>`. Currently I did not test the communication to other computers.

> In that case it will fall back to "auto" because otherwise you won't even have communication within the machine

This is a very nice idea! Still there is no communication within the machine. I had to append `<ParticipantIndex>auto</ParticipantIndex>` if I want to turn off multicast:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<CycloneDDS xmlns="https://cdds.io/config" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://cdds.io/config https://raw.githubusercontent.com/eclipse-cyclonedds/cyclonedds/master/etc/cyclonedds.xsd">
    <Domain Id="any">
        <General>
            <NetworkInterfaceAddress>wg0</NetworkInterfaceAddress>
            <AllowMulticast>false</AllowMulticast>
        </General>
        <Discovery>
            <Peers>
                <Peer address='my_computer'></Peer>
                <Peer address='robot0'></Peer>
                <Peer address='robot1'></Peer>
            </Peers>
            <ParticipantIndex>auto</ParticipantIndex>
        </Discovery>
    </Domain>
</CycloneDDS>
```ros2 run demo_nodes_cpp talker

For ROS users: The communication within the machine can be tested by using the example listener and talker:
```
ros2 run demo_nodes_cpp listener
ros2 run demo_nodes_cpp talker
```

In summary:
- there is probably a problem in `options.md`
- `ParticipantIndex` does not seem to fall back to `auto` in case multicast is disabled.

**Comment by goekce** [comment]
I also stumbled upon:

[ParticipantIndex](https://github.com/eclipse-cyclonedds/cyclonedds/blob/0bf9552ab4569a813565ff03495bf4122a14814f/docs/manual/options.md?plain=1#L123):
> - a non-negative integer less than 120, or ...

But `AutoParticipantIndex` does not have a limitation. I can open another issue if needed.
